File: backend/src/main/java/com/ll/hotel/global/aws/s3/S3Util.java
package com.ll.hotel.global.aws.s3;

import com.ll.hotel.domain.image.type.ImageType;

import java.util.UUID;

public class S3Util {
    private S3Util(){}

    // 사진 저장 path 설정
    public static String buildS3Key(ImageType imageType, long id, String fileType) {
        String fileName = UUID.randomUUID().toString() + "." + fileType;
        return switch (imageType) {
            case HOTEL -> "hotels/" + id + "/" + fileName;
            case ROOM -> "rooms/" + id + "/" + fileName;
            case REVIEW -> "reviews/" + id + "/" + fileName;
        };
    }

    public static String extractObjectKeyFromUrl(String url) {
        int domainEndIndex = url.indexOf(".com/");
        if (domainEndIndex != -1) {
            return url.substring(domainEndIndex + 5); // ".com/" 의 길이인 5를 더해줍니다
        }
        return null;
    }

    public static String getFolderPath(ImageType imageType, long id) {
        String type = switch(imageType) {
            case HOTEL -> "hotels";
            case ROOM -> "rooms";
            case REVIEW -> "reviews";
        };

        return type + "/" + id + "/";
    }
}

File: backend/src/main/java/com/ll/hotel/global/aws/s3/S3Util.kt
package com.ll.hotel.global.aws.s3

import com.ll.hotel.domain.image.type.ImageType;
import java.util.*

object S3Util {
    // 사진 저장 path 설정
    fun buildS3Key(imageType: ImageType, id: Long, fileType: String): String {
        val fileName = "${UUID.randomUUID()}.$fileType"
        return when (imageType) {
            ImageType.HOTEL -> "hotels/$id/$fileName"
            ImageType.ROOM -> "rooms/$id/$fileName"
            ImageType.REVIEW -> "reviews/$id/$fileName"
        }
    }

    // URL에서 객체 키 추출
    fun extractObjectKeyFromUrl(url: String): String? {
        val domainEndIndex = url.indexOf(".com/")
        return if (domainEndIndex != -1) {
            url.substring(domainEndIndex + 5) // ".com/"의 길이인 5를 더함
        } else {
            null
        }
    }

    // 폴더 경로 생성
    fun getFolderPath(imageType: ImageType, id: Long): String {
        val type = when (imageType) {
            ImageType.HOTEL -> "hotels"
            ImageType.ROOM -> "rooms"
            ImageType.REVIEW -> "reviews"
        }
        return "$type/$id/"
    }
}